web/src/components/TaskHeader.test.tsx

```diff
@@ -37,6 +46,17 @@ const renderTaskHeader = (task: Task, initialUserData?: UserData) =>
     </ThemeProvider>
   );
 
+const randomTradeNameLegalStructure = () => {
+  return randomInt() % 2 ? "sole-proprietorship" : "general-partnership";
+};
+
+const randomPublicFilingLegalStructure = () => {
+  const nonTradeNameLegalStructures = LegalStructures.filter(
+    (x) => x.id !== "sole-proprietorship" && x.id !== "general-partnership"
+  );
+  return nonTradeNameLegalStructures[randomInt() % nonTradeNameLegalStructures.length].id;
+};
```

Comment:
part of my rationale for this was to avoid hardcoding the logic into our tests of _which_ legal structures are public filing or trade name, since we already have that logic elsewhere

you don't have to recreate this logic, and instead could do a filter on the `LegalStructures` const that's available based on the `requiresPublicFiling` or `hasTradeName` booleans